### PR TITLE
BTP ServiceBindings & ServiceInstances

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

- Add lists and details form BTP Bindings and Instances
- I've decided to not add `userInfo` (my ignorance tells me it won't be useful) or `parametersFrom` (there was no Instance with them).
- [More info](https://gist.github.tools.sap/I509231/3caf1f406178390c57981f103eff7eed)
- ALSO fix ~tiny~ logs bug - due to Chrome (and Firefox (and Safari!)) resetting the connection every 1 minute (only on cluster), logs would double, as they weren't cleared. Fixed!
image

Testing
- Instances are on `kmain/default`.

**Related issue(s)**

### CODE REVIEW

File: core-ui/src/components/Predefined/Details/ServiceBinding/ServiceBindingData.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          onClick={() => navigateToSecret(spec.secretName)}
+        >
+          {spec.secretName}
+        </Link>
+      ),
+    },
+    { name: t('btp-service-bindings.external-name'), value: spec.externalName },
+    {
+      name: t('btp-service-bindings.binding-id'),
+      value: status.bindingID || t('common.messages.not-set'),
```
Comment (dariadomagala) [inline core-ui/src/components/Predefined/Details/ServiceBinding/ServiceBindingData.js:45]: maybe just `-` (EMPTY_TEXT_PLACEHOLDER) ?

File: core-ui/src/components/Predefined/Details/ServiceInstance/ServiceInstance.details.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      value: resource => <ServiceInstanceStatus serviceInstance={resource} />,
+    },
+  ];
+
+  return (
+    <DefaultRenderer
+      customColumns={customColumns}
+      customComponents={[
+        ServiceInstanceData,
+        ServiceBindingList,
```
Comment (dariadomagala) [inline core-ui/src/components/Predefined/Details/ServiceInstance/ServiceInstance.details.js:39]: We don't need the `service instance name` column on a service instance details view here. The user will know what instance it refers to.

File: core-ui/src/components/Predefined/Details/ServiceInstance/ServiceInstanceData.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      value: spec.externalName,
+    },
+    {
+      name: t('btp-instances.instance-id'),
+      value: status.instanceID || t('common.messages.not-set'),
+    },
+  ];
+
+  return (
+    <DefinitionList title="Instance Data" list={list} key="instance-data" />
```
Comment (dariadomagala) [inline core-ui/src/components/Predefined/Details/ServiceInstance/ServiceInstanceData.js:27]: The title should come from the i18n file. Also, imho it should be `Service Instance data`.

File: core-ui/src/components/Predefined/Details/ServiceBinding/ServiceBindingData.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import { useTranslation } from 'react-i18next';
+
+export function ServiceBindingData({ spec, status }) {
+  const { t } = useTranslation();
+  const navigateToInstance = instanceName =>
+    LuigiClient.linkManager()
+      .fromContext('namespace')
+      .navigate(`/btp-instances/details/${instanceName}`);
+
+  const navigateToSecret = secretName =>
```
Comment (dariadomagala) [inline core-ui/src/components/Predefined/Details/ServiceBinding/ServiceBindingData.js:14]: sometimes such secret doesn't exist, and it redirects the user to a non-existing page. Could we somehow distinguish if we should only show a name or a link?

File: core-ui/src/components/Predefined/List/ServiceInstances.list.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { ServiceInstanceStatus } from 'shared/components/ServiceInstanceStatus';
+
+export const ServiceInstancesList = ({ DefaultRenderer, ...otherParams }) => {
+  const { t } = useTranslation();
+
+  const customColumns = [
+    {
+      header: 'Offering name',
```
Comment (dariadomagala) [inline core-ui/src/components/Predefined/List/ServiceInstances.list.js:10]: Please use i18n translations here.

File: core-ui/src/shared/components/ServiceInstanceStatus.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { StatusBadge, Tooltip } from 'react-shared';
+
+export function ServiceInstanceStatus({ serviceInstance }) {
```
Comment (dariadomagala) [inline core-ui/src/shared/components/ServiceInstanceStatus.js:4]: cannot we join this file with Service Bindings statuses?

File: core/src/luigi-config/navigation/static-navigation-model.js [hunk 7]
```diff
@@ -618,6 +618,76 @@ export function getStaticChildrenNodesForNamespace(
...
+        '/namespaces/:namespaceId/serviceInstances?' +
+        toSearchParamsString({
+          resourceApiPath: '/apis/services.cloud.sap.com/v1alpha1',
+          readOnly: false,
+          hasDetailsView: true,
+        }),
+      viewGroup: coreUIViewGroupName,
+      keepSelectedForChildren: true,
+      context: {
+        requiredFeatures: [features.BTP_CATALOG],
```
Comment (dariadomagala) [inline core/src/luigi-config/navigation/static-navigation-model.js:637]: something doesn't work here. those two entries are shown even on a cluster without BTP resources

**Review by dariadomagala — CHANGES_REQUESTED** [review]

**Review by dariadomagala — DISMISSED** [review]

**Review by NHingerl — APPROVED** [review]

**Review by dariadomagala — APPROVED** [review]

**Pull request merged**